include/yaclib/fault/detail/condition/condition_variable.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <yaclib/fault/mutex.hpp>
+
+#include <condition_variable>
+
+namespace yaclib::detail {
+
+class ConditionVariable {
+ public:
+  ConditionVariable() noexcept = default;
+
+  ~ConditionVariable() = default;
+
+  ConditionVariable(const ConditionVariable&) = delete;
+  ConditionVariable& operator=(const ConditionVariable&) = delete;
+
+  void notify_one() noexcept;
+  void notify_all() noexcept;
+
+  void wait(std::unique_lock<yaclib::detail::Mutex>& lock) noexcept;
+
+  template <typename Predicate>
+  void wait(std::unique_lock<yaclib::detail::Mutex>& lock, Predicate predicate) {
+    auto m = lock.release();
+    auto guard = std::unique_lock<std::mutex>(m->GetImpl(), std::adopt_lock);
+    m->UpdateOwner(yaclib::detail::kInvalidThreadId);
+    YACLIB_INJECT_FAULT(_impl.wait(guard, predicate));
+    guard.release();
+    m->UpdateOwner(yaclib_std::this_thread::get_id());
+    auto new_lock = std::unique_lock(*m, std::adopt_lock);
+    lock.swap(new_lock);
```

Comment:
We need the second update for sure, since if thread comes back from waiting and tries to unlock mutex, It's state can be inconsistent since it was locked inside wait.
It seems that the first one is not needed though since I only have asserts on unlock

Reply:
Its not about second update, its about this checks

Reply:
I'm about owner checks in overall
Not only in this place

Reply:
well I believe it's not connected to fault injection and I guess whose checks somehow checked in Sanitizers for example so I guess we can omit them